scripts/leaderboard_benchmark_evaluator.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
+# Copyright (c) 2021 Intel Corporation.
+#
+# This work is licensed under the terms of the MIT license.
+# For a copy, see <https://opensource.org/licenses/MIT>.
+
+from __future__ import print_function
+
+import argparse
+import os
+import traceback
+from argparse import RawTextHelpFormatter
+
+from leaderboard.leaderboard_evaluator import LeaderboardEvaluator
+from leaderboard.utils.statistics_manager import StatisticsManager
+
+
+def serialize_header(f):
+    s =  "| Town | Route | Index | # of Vehicles | Mean FPS | Mean ratio | Sensors |\n"
+    s += "| ---- | ----- | ----- | ------------- | -------- | ---------- | ------- |\n"
+    f.write(s)
+    f.flush()
+
+
+def serialize_records(f, town, route, repetition, background_amount, duration_game, duration_system, sensors):
+
+    def serialize_sensor(sensor_spec):
+        if sensor_spec['type'] == 'sensor.camera.rgb':
+            return "RGB_{}x{}".format(sensor_spec['width'], sensor_spec['height'])
+        elif sensor_spec['type'] == 'sensor.lidar.ray_cast':
+            return "Lidar"
+        elif sensor_spec['type'] == 'sensor.other.radar':
+            return "Radar"
+        elif sensor_spec['type'] == 'sensor.other.gnss':
+            return "gnss"
+        elif sensor_spec['type'] == 'sensor.other.imu':
+            return "imu"
+        elif sensor_spec['type'] == 'sensor.opendrive_map':
+            return "map"
+        elif sensor_spec['type'] == 'sensor.speedometer':
+            return "speed"
+        else:
+            return ""
+
+    ratio = 0.0
+    if duration_system > 0:
+        ratio = duration_game / duration_system
+
+    s = "| {} | {} | {} | {} | {:03.2f} | {:03.2f} | {} |\n".format(
+        town,
+        route,
+        repetition,
+        background_amount,
+        20 * ratio,
+        ratio,
+        " ".join([serialize_sensor(s) for s in sensors]),
+    )
+    
+    f.write(s)
+    f.flush()
+
+
+class LeaderboardBenchmarkEvaluator(LeaderboardEvaluator):
+
+    def __init__(self, args, statistics_manager):
+        super(LeaderboardBenchmarkEvaluator, self).__init__(args, statistics_manager)
+
+        if os.path.isfile(args.benchmark_filename) and args.resume:
+            self.benchmark_file = open(args.benchmark_filename, "a")
+        else:
+            self.benchmark_file = open(args.benchmark_filename, "w")
+            serialize_header(self.benchmark_file)
+
+    def _register_statistics(self, config, checkpoint, entry_status, crash_message=""):
+        super(LeaderboardBenchmarkEvaluator, self)._register_statistics(
+            config, checkpoint, entry_status, crash_message
+        )
+
+        serialize_records(
```

Comment:
There is a weird interaction with the resume functionality in that crashed routes are repeated when restarting the simulation with the *resume* on. As such, that route is repeated at the records. I'd suggest checking if the crash_message is empty, to ensure that only the non-crashed routes are added to the records

Reply:
Done